### PR TITLE
ci: pass coverage env statically

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

https://discord.com/channels/428295358100013066/560503042458517505/1129158185052278854

we may get this message when fetching `CODECOV_ENV`
```
<html><head>
<meta http-equiv="content-type" content="text/html;charset=utf-8">
<title>502 Server Error</title>
</head>
<body text=#000000 bgcolor=#ffffff>
<h1>Error: Server Error</h1>
<h2>The server encountered a temporary error and could not complete your request.<p>Please try again in 30 seconds.</h2>
<h2></h2>
</body></html>
```

seems that we don't handle this error and also I think those environment variables won't change. we can just passed them.

#### Summary of Changes

pass those environment variables statically instead of a dynamic method, setting via curl.

### CODE REVIEW

**Comment by yihau** [comment]
the test build https://buildkite.com/solana-labs/solana/builds/99028
(ignore the spl test cuz I forgot to rebase when I ran it 😢)

looks all good and coverage report was uploaded successfully.

File: ci/docker-run.sh [hunk 1]
```diff
@@ -107,12 +107,23 @@ ARGS+=(
...
+  --env CODECOV_SLUG
+  --env VCS_COMMIT_ID
+  --env VCS_BRANCH_NAME
+  --env VCS_PULL_REQUEST
+  --env VCS_SLUG
+  --env VCS_TAG
+  --env CI_BUILD_URL
+  --env CI_BUILD_ID
+  --env CI_JOB_ID
+)
```
Comment (ryoqun) [inline ci/docker-run.sh:126]: I'd like avoid hard-coding as much as possible. i know i'm opiniated and seldom these envs should change. but i just don't like hard-coding....

seems `curl --fail` does the trick.


```
$ curl -sS --retry 5 --retry-delay 2 --retry-connrefused https://codecov.io/env
<html><head>
<meta http-equiv="content-type" content="text/html;charset=utf-8">
<title>502 Server Error</title>
</head>
<body text=#000000 bgcolor=#ffffff>
<h1>Error: Server Error</h1>
<h2>The server encountered a temporary error and could not complete your request.<p>Please try again in 30 seconds.</h2>
<h2></h2>
</body></html>
<html><head>
<meta http-equiv="content-type" content="text/html;charset=utf-8">
<title>502 Server Error</title>
</head>
<body text=#000000 bgcolor=#ffffff>
<h1>Error: Server Error</h1>
<h2>The server encountered a temporary error and could not complete your request.<p>Please try again in 30 seconds.</h2>
<h2></h2>
</body></html>
#!/usr/bin/env bash

# Apache License Version 2.0, January 2004
# https://github.com/codecov/codecov-bash/blob/master/LICENSE

```

```
$ curl --fail -sS --retry 5 --retry-delay 2 --retry-connrefused https://codecov.io/env
#!/usr/bin/env bash

# Apache License Version 2.0, January 2004
# https://github.com/codecov/codecov-bash/blob/master/LICENSE

set -e +o pipefail

VERSION="1.0.6"  

add()
{
  if [ -z "$1" ];
  then
    return
  fi

```

Comment (ryoqun) [reply ci/docker-run.sh:126]: the problem is that `curl` is printing 502 response body. blame flaky codecov infra.
seems their infra changed when i added this at #8263. so blame me as well. ;)

**Review by ryoqun — CHANGES_REQUESTED** [review]
thanks for debugging this. however, i want different approach is taken..

File: ci/docker-run.sh (hunk 1 above)
Comment (yihau) [reply ci/docker-run.sh:126]: I do this cuz
1. we can see what we're passing easily also make it controllable.
2. although we pass those env dynamically, we still need to set them manually to machines or we get an empty env in the CI environment

What's the benefit you would like to keep the dynamic environment variable set

Comment (ryoqun) [reply ci/docker-run.sh:126]: (sorry i have to go to chore soon)

> What's the benefit you would like to keep the dynamic environment variable set

i'd say a maintenance burden. for example, this pr starts to miss to export some of these envs, which is used for codecov report metadata...

```
elif [ "$CI" = "true" ] && [ "$BUILDKITE" = "true" ];
then
  add "CI"
  add "BUILDKITE"
  add "BUILDKITE_BRANCH"
  add "BUILDKITE_BUILD_NUMBER"
  add "BUILDKITE_JOB_ID"
  add "BUILDKITE_BUILD_URL"
  add "BUILDKITE_PROJECT_SLUG"
  add "BUILDKITE_COMMIT"

```

also, codocov doc is the recommended way to `curl`...

Comment (yihau) [reply ci/docker-run.sh:126]: true. it is easy to miss some needed environment variables. just opened another solution for handling curl response code. #32492

**Comment by yihau** [comment]
use #32494

**Pull request closed without merging**